rag/utils/s3_conn.py

```
@@ -113,15 +113,15 @@ def bucket_exists(self, bucket, *args, **kwargs):
-            self.conn[0].create_bucket(Bucket=bucket)
-            logging.debug(f"create bucket {bucket} ********")
-
-        r = self.conn[0].upload_fileobj(BytesIO(binary), bucket, fnm)
-        return r
+        try:
+            if self.bucket:
+                self.conn[0].head_bucket(Bucket=self.bucket)
+            else:
+                self.conn[0].list_buckets()
+            return True
+        except Exception as e:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)